client/jetpack-connect/signup.js

```diff
@@ -212,7 +260,111 @@ export class JetpackSignup extends Component {
 			</LoggedOutFormLinks>
 		);
 	}
+
+	renderWooDna() {
+		const { authQuery, isFullLoginFormVisible, translate, usernameOrEmail } = this.props;
+		const { isCreatingAccount, signUpUsernameOrEmail } = this.state;
+		let header, subHeader, content;
+		const footerLinks = [];
+		const email = signUpUsernameOrEmail || usernameOrEmail || authQuery.userEmail;
+		const wooDna = this.getWooDnaConfig();
+		let pageTitle;
+
+		if ( this.state.showWooDnaLoginForm ) {
+			if ( isFullLoginFormVisible ) {
+				header = translate( 'Log in to your WordPress.com account' );
+				/* translators: pluginName is the name of the Woo extension that initiated the connection flow */
+				subHeader = translate(
+					'Your account will enable you to start using the features and benefits offered by %(pluginName)s',
+					{
+						args: {
+							pluginName: wooDna.name( translate ),
+						},
+					}
+				);
+				pageTitle = translate( 'Login to WordPress.com' );
+				footerLinks.push(
+					<LoggedOutFormLinkItem key="signup" onClick={ this.showWooDnaSignupView }>
+						{ this.props.translate( 'Create a new account' ) }
+					</LoggedOutFormLinkItem>
+				);
+				footerLinks.push(
+					<LoggedOutFormLinkItem
+						key="lostpassword"
+						href={ addQueryArgs(
+							{ action: 'lostpassword' },
+							login( { locale: this.props.locale } )
+						) }
+					>
+						{ this.props.translate( 'Lost your password?' ) }
+					</LoggedOutFormLinkItem>
+				);
+			} else {
+				header = wooDna.name( translate );
+				subHeader = translate( 'Enter your email address to get started' );
+				pageTitle = translate( 'Connect' );
+			}
+		} else {
+			header = wooDna.name( translate );
+			subHeader = translate( 'Create an account' );
+			pageTitle = translate( 'Create a WordPress.com account' );
+			footerLinks.push(
+				<LoggedOutFormLinkItem key="login" onClick={ () => this.showWooDnaLoginView() }>
+					{ this.props.translate( 'Log in with an existing WordPress.com account' ) }
+				</LoggedOutFormLinkItem>
+			);
+		}
+
+		footerLinks.push(
+			<LoggedOutFormLinkItem key="back" href={ authQuery.redirectAfterAuth }>
+				<Gridicon size={ 18 } icon="arrow-left" />{ ' ' }
+				{
+					// translators: eg: Return to The WordPress.com Blog
+					this.props.translate( 'Return to %(sitename)s', {
+						args: { sitename: decodeEntities( authQuery.blogname ) },
+					} )
+				}
+			</LoggedOutFormLinkItem>
+		);
+		const footer = <LoggedOutFormLinks>{ footerLinks }</LoggedOutFormLinks>;
+
+		if ( this.state.showWooDnaLoginForm ) {
```

Comment:
Why do we need two separate if blocks checking for `this.state.showWooDnaLoginForm`?

Reply:
I thought it made the code a bit more readable, but that's dubious. I've just changed it in https://github.com/Automattic/wp-calypso/pull/41798/commits/a49c0296104aa552dd3552be175f2eee2612287d, is that better or worse?